feat: add option early return in case of error

thenicolau opened the pull request:
## Fixes Or Enhances


**Make sure that you've checked the boxes below before you submit PR:**
- [X] Tests exist or have been written that cover this particular change.

@go-playground/validator-maintainers

thenicolau commented:
[Issue - 1371](https://github.com/go-playground/validator/issues/1371)

deankarn commented:
Asking about the use case here to learn if it's worth saving a few nanoseconds :) https://github.com/go-playground/validator/issues/1371#issuecomment-2660989420

thenicolau commented:
I noticed that this old PR had a merge problem, and the issue linked to it was also updated, so I fixed the issues and it is now ready to be reviewed again

zemzale reviewed (approved):
Just as mentioned before I am not sure about the use case. Not that it's invalid, but more if we actually want to add this to the library. 

But the code looks good, so approving the code change
